utils.py

```diff
@@ -891,11 +891,22 @@ def check_auth(user, hutch):
 def check_special(ioc, hutch):
     lines = open(SPECIAL_FILE % hutch).readlines()
     lines = [l.strip() for l in lines]
-    for l in lines:
+    for l in lines: #Note: (josh) as long as the ioc name is in the special file entry we let it be cycled by any user 
         if l == ioc:
             return True
     return False
 
+# checks if ioc is marked as toggleable between defined versions of IOCs
+# schema will be ioc_name:permittedversion1,permittedversion2,etc
+def check_special_toggle(ioc, hutch, version):
+    lines = open(SPECIAL_FILE % hutch).readlines()
```

Comment:
It's bad practice to not close out your files.  Python's `open` function can be used as a context manager to do this.  (I realize this is how it's done in other parts of this codebase, those should be changed too)

```python
with open(filepath, 'r') as file:
    lines = file.readlines()
```

Reply:
Agreed.  Feel free to change this.

Reply:
totally agree with the when you touch something make it better ideology, even when that thing is python2.7

Reply:
theres a real refactor where check_special gets overloaded to take a third argument of version and then we only have to do one piece of file i/o on this special auth file, but I would probably recommend upgrading to python3 before adding wrappers into this codebase

Reply:
I'm not against check_special having a third argument, as long as you read the file each time you check a condition.  (The file could change... you really *do* have to re-read it each time.)

This was sort of my comment about two lines above... do you want "a: b,c" to mean you are allowing toggles *only*, or does it allow enabling/disabling *and* toggles?

Reply:
Understood. Pushing up a change that:
a. Closes file, while I have forgotten most python2 but I think `with` will work as I expect.
b. Makes sure user has ability to disable / enable to ioc theyre trying to toggle

I am choosing to not change the current check_special function even though it's only invoked in one place to keep my impact to this effectively stagnant codebase minimal. Let me know if that perspective is off.

Reply:
Oh geez it totally didn't register to me that this was python 2.7.  I'm not sure when the context manager was added, but suspect it wasn't until after python 3